### ISSUE TITLE
Plugin crashing with multiple custom rules and icons

### ISSUE THREAD

**Issue by Caotenhoch2** [FlorianWoelki/obsidian-iconize#413]
**Describe the bug**
The plugin is not loading correctly when opening obsidian and not displaying any icons and settings. This can be fixed by deleting the data.json. Since this file has not externally been tampered with by me, an error must have occurred when writing/reading it. 

The error in the developer console:
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'iconColor')
    at IconFolderPlugin.getIconColor (plugin:obsidian-icon-folder:6059:25)
    at Object.addAll (plugin:obsidian-icon-folder:4424:46)
    at IconFolderPlugin.handleChangeLayout (plugin:obsidian-icon-folder:5759:14)
    at eval (plugin:obsidian-icon-folder:5582:57)
    at t.onLayoutReady (app.js:1:2205166)
    at IconFolderPlugin.eval (plugin:obsidian-icon-folder:5582:32)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-icon-folder:31:58)
```

**To Reproduce**
1. Replace the data.json with this [data.json](https://github.com/FlorianWoelki/obsidian-iconize/files/14812367/data.json)
 (Which I created only using the settings)
2. Restart obsidian

**Expected behavior**
The plugin should load correctly and display icons

**Comment by FlorianWoelki** [comment]
thanks for reporting the issue!

this is probably already fixed with https://github.com/FlorianWoelki/obsidian-iconize/commit/650da87a74180edeb5d91afb5eb16e6a3b4e251a. i'll make a new release tomorrow.

**Comment by FlorianWoelki** [comment]
This issue should now have been resolved in [v2.11.5](https://github.com/FlorianWoelki/obsidian-iconize/releases/tag/2.11.5).
